From: Ruben Undheim <ruben.undheim@gmail.com>
Date: Mon, 19 Sep 2016 08:03:15 +0200
Subject: Needed to make sure that CXXFLAGS are propagated from the environment

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Makefile
+++ b/Makefile
@@ -6,11 +6,11 @@
 # build optimized without -DNDEBUG
 # OPTDEBUGFLAGS = -O0 -fno-inline -g
 # OPTDEBUGFLAGS = -O3 -DNDEBUG
-OPTDEBUGFLAGS = -O2
+OPTDEBUGFLAGS ?= -MD -O2
 SRC = src
 
 # clang only: -Wglobal-constructors
-CXXFLAGS = -I$(SRC) -std=c++11 -MD $(OPTDEBUGFLAGS) -Wall -Wshadow -Wsign-compare -Werror
+CXXFLAGS += -I$(SRC) -std=c++11 $(OPTDEBUGFLAGS) -Wall -Wshadow -Wsign-compare -Werror
 LIBS = -lm
 
 DESTDIR = /usr/local
